### PR TITLE
chore: add a deadlink checker for nav/sidebar items

### PR DESCRIPTION
**Pull request by mujahidkay**
Closes: https://github.com/Agoric/documentation/issues/1225

Added this check to build script to keep it consistent with vitepress's built-in deadlink checker for md files (raises error during build time)

Unrelated but also formats the scripts dir.

Testing:

You can test by checking out the branch and entering invalid links.

### CODE REVIEW

File: scripts/checkLinks.mjs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const extractLinks = (content) => {
+  const linkRegex = /link:\s*'([^']+)'/g;
```
Comment (amessbee) [inline scripts/checkLinks.mjs:9]: I guess there is nothing to stop someone from using double quotes with links?

**Comment by mujahidkay** [comment]
> @mujahidkay wondering whether we have any other options to fix #1225 ?

I think [Linkinator](https://www.npmjs.com/package/linkinator) seems like another option but I haven't really tested it. I will give it a go in my free time.

File: scripts/checkLinks.mjs (hunk 1 above)
Comment (mujahidkay) [reply scripts/checkLinks.mjs:9]: good catch!

Comment (amessbee) [reply scripts/checkLinks.mjs:9]: I think this would also break the CI if there is a comment that contains a broken link?

**Review by amessbee — APPROVED** [review]
lgtm - nice work!

**Pull request merged**